src/format/mod.rs

```diff
@@ -386,10 +446,11 @@ pub fn format_item<'a>(
     time: Option<&NaiveTime>,
     off: Option<&(String, FixedOffset)>,
     item: &Item<'a>,
-) -> fmt::Result {
+) -> FormatResult<()> {
```

Comment:
This should continue returning a regular `fmt::Result`

Reply:
There will be duplicated code if we do that. Or I can use fmt::Result everywhere but then we have no clear error message when the locale that was provided is unknown or not. Are you ok with that?

Reply:
it's fine to have a bit of duplicated code, but if the feature set is the same shouldn't it be guaranteed that the only possible error in `format_inner` is the `FormatResult::Format`? The other error variants can only come from locale handling?

Reply:
No actually you have a few more errors that are not fmt::Error:
https://github.com/chronotope/chrono/blob/febb8dc168325ac471b54591c925c48b6a485962/src/format/mod.rs#L492
https://github.com/chronotope/chrono/blob/febb8dc168325ac471b54591c925c48b6a485962/src/format/mod.rs#L643
https://github.com/chronotope/chrono/blob/febb8dc168325ac471b54591c925c48b6a485962/src/format/mod.rs#L647

Since I made a Locale enum as you suggested, the local handling is not done in chrono anymore so it is infallible (well, at least here but I think we still should re-export this Locale enum). I reverted the code to fmt::Error.

(I will continue tomorrow, I have not done the re-export yet)